shuup/campaigns/models/basket_conditions.py

```diff
@@ -209,3 +209,30 @@ def values(self):
     @values.setter
     def values(self, values):
         self.contacts = values
+
+
+class CategoryProductsBasketCondition(BasketCondition):
+    identifier = "basket_category_condition"
+    name = _("Category products in basket")
+
+    operator = EnumIntegerField(
+        ComparisonOperator, default=ComparisonOperator.GTE, verbose_name=_("operator"))
+    quantity = models.PositiveIntegerField(default=1, verbose_name=_("quantity"))
+    category = models.ForeignKey(Category, verbose_name=_("category"), blank=True)
+
+    def matches(self, basket, lines):
+        product_id_to_qty = get_product_ids_and_quantities(basket)
+        category_product_ids = self.category.shop_products.filter(
+            product_id__in=product_id_to_qty.keys()
+        ).values_list(
+            "product_id", flat=True
+        )
+        product_count = sum(product_id_to_qty[product_id] for product_id in category_product_ids)
+        if self.operator == ComparisonOperator.EQUALS:
+            return bool(self.quantity == product_count)
+        else:
+            return bool(self.quantity <= product_count)
```

Comment:
Strange that here this comparison is being flipped. In my opinion, it would be easier to read this by comparing `product_count >= self.quantity)` since the operator itself says `GTE` not `LTE` :) `ProductsInBasketCondition` does the `count >= self.quantity)` so this should follow the same pattern IMO.